components/EditButton.js

```diff
@@ -0,0 +1,7 @@
+export default function EditButton({ showModal, data }) {
+  return (
+    <button type="button" onClick={() => showModal.toggle("edit", data)}>
```

Comment:
you could pass a generic onClick function this would improve the simplicity: 

```js
function EditButton({onClick}) {
  return <button onClick={onClick} ...
}

<EditButton onClick={() => showModal.toggle('edit',data)}>
```